openedx/core/djangoapps/safe_sessions/tests/test_middleware.py

```diff
@@ -188,30 +201,90 @@ def assert_response_with_delete_cookie(
             self.assert_response(set_request_user=set_request_user, set_session_cookie=set_session_cookie)
             assert mock_delete_cookie.called == expect_delete_called
 
-    def test_success(self):
+    @patch("django.core.cache.cache.set")
+    @patch("django.core.cache.cache.touch")
+    def test_success(self, mock_cache_set, mock_cache_touch):
```

Comment:
I'm curious if we could test the consequences of this, rather than the cache value itself? For example, using multiple calls with or without a mismatch and showing that logging happens where and when we expect. Food for thought.